crates/provider/src/shared/provider.rs

```diff
@@ -0,0 +1,24 @@
+use std::error::Error;
+
+use async_trait::async_trait;
+
+use super::types::{NativeRunCommandOptions, PodDef, RunCommandResponse};
+
+#[async_trait]
+#[allow(non_upper_case_globals)]
+pub trait Provider {
+    fn create_namespace(&mut self) -> Result<(), Box<dyn Error>>;
+    fn setup_cleaner(&self) -> Result<(), Box<dyn Error>>;
```

Comment:
Here you can provide the default implementation that @l0r1s  mention [here](https://github.com/paritytech/zombienet-sdk/pull/58/files#r1213485634)